### ISSUE TITLE
lll is not disabled for --enable-all --disable lll

### ISSUE THREAD

**Issue by willfaught** [alecthomas/gometalinter#194]
```
$ gometalinter --enable-all --disable lll --debug ./...
DEBUG: PATH=/Users/will.faught/.rbenv/shims:/Users/will.faught/.nvm/versions/node/v0.12.4/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/opt/X11/bin:/Users/will.faught/Developer/go/bin:/usr/local/opt/fzf/bin:/Users/will.faught/Developer/go/bin
DEBUG: GOPATH=/Users/will.faught/Developer/go
DEBUG: GOBIN=
DEBUG: linting path .
DEBUG: linting path grpc/error
DEBUG: linting path grpc/fortune
DEBUG: linting path grpc/listing
DEBUG: linting path mocks
DEBUG: linting with ineffassign: ineffassign -n {path} (on .)
DEBUG: linting with gas: gas -fmt=csv {path}/*.go (on grpc/error)
DEBUG: linting with interfacer: interfacer {path} (on ./...)
DEBUG: linting with lll: lll -g -l {maxlinelength} {path}/*.go (on .)
[snip]
```

See the last line of the output.

Is this how I'm supposed to run all but lll?

**Comment by alecthomas** [comment]
Since the switch to a configuration file (in 1df416899120d87ae62df60322721e5516615afb), there was no concept of ordering.

**Comment by willfaught** [comment]
Is there no brief, concise way to use this particular set (all but lll) of
linters?

**Comment by alecthomas** [comment]
Sorry, I should have been clear: that commit that closed this issue re-enables that behaviour. So yes, it works again.

**Comment by johanbrandhorst** [comment]
This change broke the behaviour of using a config file with explicitly disabled linters. Now you have to disable them on the command line instead?

**Comment by alecthomas** [comment]
You can DisableAll, then Enable the ones you want. The problem with having both an enable and a disable list in a configuration file is which takes precedence, enable or disable?

**Comment by alecthomas** [comment]
Ah, I see... I removed DisableAll... I'll add it back in.

**Comment by johanbrandhorst** [comment]
Indeed, unfortunately you will break someones workflow, either the ones using `Enable: {}` or the ones using `Disable: {}`. We used `Disable: {}` so here I am :(.

**Comment by willfaught** [comment]
@alecthomas I'm confused. Can I accomplish what I want currently with `--enable-all --disable lll` as flags, or in a file, or neither? If it's not possible, is that equivalent to the flags `--enable testify --enable test --enable gofmt --enable goimports --enable misspell --enable unused`?

**Comment by alecthomas** [comment]
@willfaught I'm not sure why you're confused, I already said your original issue was fixed. Subsequent discussion has been about a regression in config file behaviour.

**Comment by alecthomas** [comment]
@johanbrandhorst The current behaviour is that no linters are enabled unless they are in the `Enable` list. So, basically you have to set exactly what linters you want to run. This is both simpler, and deterministic.

**Comment by johanbrandhorst** [comment]
@alecthomas a problem with that is that you can't automatically enable new linters, which I was hoping of achieving by only explicitly disabling linters. This way users of this library who may want to add new linters as they are added here will need to keep up to date with all commits here rather than have new linters automatically enabled by virtue of not being explicitly disabled. Could we get some way to support that workflow?

**Comment by alecthomas** [comment]
FYI a Disable configuration option was implemented by @pocke in c5e7d73d2e4e4a33af20af87a7476b941c537e76. Thanks!
